src/vsphere_cpi/spec/integration/nsxt_spec.rb

```diff
@@ -198,6 +200,70 @@
         end
       end
     end
+
+    context 'when load balancers are specified' do
+      let(:port_no) { '443' }
+      let(:vm_type) do
+        {
+          'ram' => 512,
+          'disk' => 2048,
+          'cpu' => 1,
+          'nsxt' => {
+            'lb' => {
+                'server_pools' => [
+                  {
+                    'name' => server_pool_name_1,
+                    'port' => port_no
+                  },
+                  {
+                    'name' => server_pool_name_2,
+                    'port' => 80
+                  }
+                ]
+              }
+            }
+          }
+      end
+
+      context 'but atleast one server pool does not exist' do
+        it 'raises an error' do
+          expect do
+            simple_vm_lifecycle(cpi, @nsxt_opaque_vlan_1, vm_type)
+          end.to raise_error(VSphereCloud::ServerPoolsNotFound)
+        end
+      end
+      context 'and all server pool exists' do
+        let(:nsgroup_1) { create_nsgroup(nsgroup_name_1) }
+        let(:server_pool_1) { create_server_pool(server_pool_name_1) }
+        let(:server_pool_2) { create_server_pool(server_pool_name_2, nsgroup_1) }
+
+        before do
+          expect(server_pool_1).to_not be_nil
+          expect(server_pool_2).to_not be_nil
+        end
+
+        after do
+          delete_server_pool(server_pool_1)
+          delete_server_pool(server_pool_2)
+          delete_nsgroup(nsgroup_1)
+        end
+
+        it 'adds vm to existing static server pools and adds all logical ports of the VM to NSGroups associated with the dynamic server pool' do
+          simple_vm_lifecycle(cpi, @nsxt_opaque_vlan_1, vm_type) do |vm_id|
+            vm = cpi.vm_provider.find(vm_id)
+            expect(vm).to_not be_nil
```

Comment:
These aren't actually necessary in a test. If they're null we'll just get exceptions thrown which will also fail the example.